test/unit/discovery/CloudDiscoveryInvalidTokenTest.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const { expect } = require('chai');
+const {HazelcastCloudDiscovery} = require("../../../lib/discovery/HazelcastCloudDiscovery");
+const { HazelcastError } = require('../../../lib');
+const { ClientConfigImpl } = require('../../../lib/config/Config');
+const {getRejectionReasonOrThrow} = require("../../TestUtil");
+
+describe('Cloud Discovery Invalid Token Test', function () {
+    it('Cloud Discovery should throw proper error with an invalid token', async function () {
+        const clientConfig = new ClientConfigImpl();
+        const cloudDiscovery = new HazelcastCloudDiscovery(
+            clientConfig.properties['hazelcast.client.cloud.url'], 2 << 31
+        );
+
+        try {
```

Comment:
This part returns `TypeError: Cannot read properties of undefined (reading 'endpointUrl')`